### ISSUE TITLE
Error Failed to get your identity

### ISSUE THREAD

**Issue by YueMiyuki** [Erisa/discord-oidc-worker#13]
![image](https://github.com/Erisa/discord-oidc-worker/assets/76854136/51115452-3719-4107-b93c-80e12162c6c1)

**Comment by kms0219kms** [comment]
<img width="1792" alt="Screenshot 2024-03-24 at 11 57 19 AM" src="https://github.com/Erisa/discord-oidc-worker/assets/67222970/0ce72259-3bbd-4501-a79e-917c6af0ebaa">
I had a similar error and I have no idea to fix it...

**Comment by YueMiyuki** [comment]
Still doesnt work... I made sure again that the secret is correct...

Any fix?

**Comment by kms0219kms** [comment]
I've fixed this issue by resetting the client secret. Can you try this?

**Comment by YueMiyuki** [comment]
Tried, still no luck...

**Comment by kms0219kms** [comment]
@YueMiyuki, Can you try with https://github.com/lunaiz-corp/discord-oidc-worker this?
I recently forked this repository and completely rewrittened to h3.

**Comment by kms0219kms** [comment]
Thanks so much for the stars! I just released a new feature (role data cache + new scope (which already tried to add on #4)) that includes a bug fix that could be causing the issue you're experiencing. (I had the same bug as you a few days ago, so I'm in the process of rewriting everything in typescript with h3.)

**Comment by Erisa** [comment]
Do you know what caused/causes this? In my usage and others I've seen use my original Worker, I've not seen this be an issue so curious where the "bug" is

**Comment by kms0219kms** [comment]
> Do you know what caused/causes this? In my usage and others I've seen use my original Worker, I've not seen this be an issue so curious where the "bug" is

I couldn't figure this out either. It failed because it only returned 400 or 500 without any errors or logs, and Cloudflare says it failed because the worker returned a status other than 200. But I thought it was an unknown and incorrect behavior, like a conflict with another code, so I rewrote everything.

After the rewrite, I tried to prevent the wrong behavior like a conflict as much as possible by applying Clean Architecture and clearly separating all code. Since the problem is not happening now, I just suggested that try using my fork and see if it is the same.

**Comment by Erisa** [comment]
Okay thanks for sharing anyway, it could have been due to the code changes or due to just setting up the config again from scratch.

**Comment by kms0219kms** [comment]
Yes, I agree. Since we applied Clean Architecture, we can add more detailed and accurate logs. If the error situation is reproduced, I will try to print the log again.

@YueMiyuki, if you get the same error, please let me know. I'll try to get some logs.

**Comment by bloodcloak** [comment]
@kms0219kms I tried using your fork but still encounter the same error.

Enabling the worker logging, the error code that is shown after the POST to the /token endpoint is 

![Image](https://github.com/user-attachments/assets/33d1bc09-4fb3-4d7d-9914-d53302f3d0c6)

The secret was put to the wrangler environment with and without quotes surrounding the value. I also did the same to the Auth configuration in the OID settings with the same error.

Do I need to include quotes around the client secret or it can be the direct value from the Discord Oauth2 Dashboard when I run wrangler secret put?

I also tried resetting the client secret token and still get the same error.

**Comment by kms0219kms** [comment]
@bloodcloak Interesting. Can you just send me a DM so I can investigate more? My Discord info can be found from my Github profile.

**Comment by YueMiyuki** [comment]
I can no longer reproduce anymore on latest version...
But rolling back to an old version, I can reproduce this

**Comment by YueMiyuki** [comment]
> [@kms0219kms](https://github.com/kms0219kms) I tried using your fork but still encounter the same error.
> 
> Enabling the worker logging, the error code that is shown after the POST to the /token endpoint is
> 
> ![Image](https://github.com/user-attachments/assets/33d1bc09-4fb3-4d7d-9914-d53302f3d0c6)
> 
> The secret was put to the wrangler environment with and without quotes surrounding the value. I also did the same to the Auth configuration in the OID settings with the same error.
> 
> Do I need to include quotes around the client secret or it can be the direct value from the Discord Oauth2 Dashboard when I run wrangler secret put?
> 
> I also tried resetting the client secret token and still get the same error.

Any chance you can share the detailed error message with me?
My discord is miyuki.yue_3747

**Comment by bloodcloak** [comment]
So I did more digging and traced it down to a configuration issue with how I setup the KV namespace. 

The error being raised at this line here

![Image](https://github.com/user-attachments/assets/272d8a96-36d2-4233-adba-fcf1c6c5fe0e)

When I created the namespace I used a different name than `KV` as shown below.

![Image](https://github.com/user-attachments/assets/cca984dc-fa49-4693-be17-b4bba92dd2ff)

Creating a new namespace with the name `KV` and using that ID resolved the issue. 

![Image](https://github.com/user-attachments/assets/20089c5b-9d3c-4fae-addd-352c74eb6e09)

Would suggest clarification on the readme instructions for this step / add a picture like above

**Comment by YueMiyuki** [comment]
> So I did more digging and traced it down to a configuration issue with how I setup the KV namespace.
> 
> The error being raised at this line here
> 
> ![Image](https://github.com/user-attachments/assets/272d8a96-36d2-4233-adba-fcf1c6c5fe0e)
> 
> When I created the namespace I used a different name than `KV` as shown below.
> 
> ![Image](https://github.com/user-attachments/assets/cca984dc-fa49-4693-be17-b4bba92dd2ff)
> 
> Creating a new namespace with the name `KV` and using that ID resolved the issue.
> 
> ![Image](https://github.com/user-attachments/assets/20089c5b-9d3c-4fae-addd-352c74eb6e09)
> 
> Would suggest clarification on the readme instructions for this step / add a picture like above

Hi.
I don't think that's where the issue came out.
In my case, rolling back to the old version still reproduce the error, but updating to latest doesn't.
(Using same KV namespace and worker)

**Comment by kms0219kms** [comment]
@bloodcloak That should not be the case, as I always create the namespace the KV as `Cloudflare Access - Discord OAuth KV` or something kind of like this... 

@YueMiyuki I sent the friend req, can you check it?

It might be helpful to get the error details or log details as much as you can, as I am very interested in to fix this.

**Comment by YueMiyuki** [comment]
> @bloodcloak That should not be the case, as I always create the namespace the KV as `Cloudflare Access - Discord OAuth KV` or something kind of like this... 
> 
> @YueMiyuki I sent the friend req, can you check it?
> 
> It might be helpful to get the error details or log details as much as you can, as I am very interested in to fix this.

Hi 
I just accepted you friend request :)

**Comment by 4m2c4bl3** [comment]
nevermind, i solved my version of the error. my issue was copy-pasting urls in my identity providers, only the auth one has /authorize/ in it
